File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/EmployeeController.cs
using System.Web.Mvc;
using MvcApplication1.Models;

namespace MvcApplication1.Controllers
{
    public class EmployeeController : Controller
    {
        private EmployeeRepository _repository = new EmployeeRepository();

        // GET: /Employee/
        public ActionResult Index()
        {
            return View();
        } 

        // GET: /Employee/Create
        [AcceptVerbs(HttpVerbs.Get)]
        public ActionResult Create()
        {
            return View();
        } 

        // POST: /Employee/Create
        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(Employee employeeToCreate)
        {
            try
            {
                _repository.InsertEmployee(employeeToCreate);
                return RedirectToAction("Index");
            }
            catch
            {
                return View();
            }
        }

        // POST: /Employee/Delete/1
        [AcceptVerbs(HttpVerbs.Delete)]
        public ActionResult Delete(int id)
        {
            _repository.DeleteEmployee(id);
            return Json(true);
        }

    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/EmployeeController.vb
Public Class EmployeeController
    Inherits System.Web.Mvc.Controller

    Private _repository As New EmployeeRepository()

    ' GET: /Employee/Create
    Function Index() As ActionResult
        Return View()
    End Function


    ' GET: /Employee/Create
    <AcceptVerbs(HttpVerbs.Get)> _
    Function Create() As ActionResult
        Return View()
    End Function

    ' POST: /Employee/Create
    <AcceptVerbs(HttpVerbs.Post)> _
    Function Create(ByVal employeeToCreate As Employee) As ActionResult
        Try
            _repository.InsertEmployee(employeeToCreate)
            Return RedirectToAction("Index")
        Catch
            Return View()
        End Try
    End Function

    ' DELETE: /Employee/Create
    <AcceptVerbs(HttpVerbs.Delete)> _
    Function Delete(ByVal id As Integer) As ActionResult
        _repository.DeleteEmployee(id)
        Return Json(True)
    End Function


End Class
